src/clover/optimisation/appraisal.py

```diff
@@ -236,8 +248,46 @@ def _simulation_environmental_appraisal(  # pylint: disable=too-many-locals
         round(total_system_ghgs, 3),
     )
 
+def _get_grid_pricing_tier(
+    grid_energy:float,
+    tiers: Dict[str, Any],
+    daily_peak_demand: float,
+    exchange_rate: float,
+):
+#run on a daily basis # should this be a for loop? every month over 20 years
+
+    if grid_type=='dieselg':
+        if household_daily_peak_demand<= 1.1: #A to kW and because peak so kWh so here I am saying if at anytime during a day the peak demand <=5A power then we only need tier 5A
+            tier_i_am_in= grid_type#upper_bound-consumption_5
+            #subscription_cost=(tier_i_am_in.subscription_cost)/exchange_rate
+            # I want the subscription costs to add them to the total_cost
+            #costs=(tier_i_am_in.cost)/exchange_rate
+        else:
+            tier_i_am_in= grid_type#upper_bound-consumption_10
+            #subscription_cost=(tier_i_am_in.subscription_cost)/exchange_rate
+            #costs=(tier_i_am_in.cost)/exchange_rate
+
+#run on a monthly basis # for loop monthly over 20 years
+
+    if grid_type=='edl':
```

Comment:
# :bug: Hardcoding
Similarly, we don't want, or need, these hard-coded strings.

The way we've set this up:
* we have a grid,
* it has some tiers,
* and these say at what threshold they're active, and what cost they are.

So... we can just work out which tier we're in based on the number in the tier, without needing this `if this, if this, if this, ...` block 😄

Reply:
You could try something like:
```
# Filter out based on whether the grid is current drawing or max-power in its pricing
if grid.type == GridType.CURRENT_DRAW:
    # Determine the peak current
    # Compare this to the tariff numbers, possibly with some dictionary
elif grid.type == GirdType.DAILY_POWER:
    # Sum over the time period, you will need to code this somewhere, here would be fine for now
    # Determine the energy that was consumed
    # Compare this, again, with some kind of dictionary
else:
    raise Exception(
        "Grid type must be one of {}".format(
            ", ".join({e.value for e in GridType}) # This will print all the allowed values
        )
    )

Reply:
Okay understood what's happening here... therefore my code should be looking like:
```
def _get_grid_pricing_tier(
    grid_energy:float,
    tiers: Dict[str, Any],
    daily_peak_demand: float,
    exchange_rate: float,
):
# Filter out based on whether the grid is current drawing or max-power in its pricing
 if grid.type == GridType.CURRENT_DRAW: #DIESEL GENERATOR
    #Determine the peak current
    if household_daily_peak_demand<= tier_5A_supply: #A to kW and because peak so kWh so here I am saying if at anytime during a day the peak demand <=5A power then we only need tier 5A
        tier_i_am_in= grid_type#upper_bound-consumption_5
    else:
        tier_i_am_in= grid_type #upper_bound-consumption_10
    # Compare this to the tariff numbers, possibly with some dictionary
 elif grid.type == GridType.DAILY_POWER: #EDL
    # Sum over the time period, you will need to code this somewhere, here would be fine for now
    # Determine the energy that was consumed
    if household_monthly_demand<= upper_bound_tier_edl_1:
        tier_i_am_in #is the tier with ["upper_bound"]["consumption"]=100
    elif household_monthly_demand<= upper_bound_tier_edl_1 & household_monthly_demand<= upper_bound_tier_edl_2:
        tier_i_am_in #is the tier with ["upper_bound"]["consumption"]=200
    elif household_monthly_demand<= upper_bound_tier_edl_2 & household_monthly_demand<= upper_bound_tier_edl_3:
        tier_i_am_in #is the tier with ["upper_bound"]["consumption"]=300
    elif household_monthly_demand<= upper_bound_tier_edl_3 & household_monthly_demand<= upper_bound_tier_edl_4:
        tier_i_am_in #is the tier with ["upper_bound"]["consumption"]=400
    elif household_monthly_demand<= upper_bound_tier_edl_4 &  household_monthly_demand<= upper_bound_tier_edl_5:
        tier_i_am_in #is the tier with ["upper_bound"]["consumption"]=1000
 else:
    raise Exception(
        "Grid type must be one of {}".format(
            ", ".join({e.value for e in GridType}) # This will print all the allowed values
        )
    )

```

I am a bit struggling to convert my text into code, in other words, creating dictionary for the values that I want to compare... As said before an example could be very helpful